{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# バクテリアゲノム解析"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "このファイルは bash カーネルを使用しています。bash カーネルをインストールしていない場合は、ターミナルで\n",
    "```\n",
    "pip install bash_kernel\n",
    "python -m bash_kernel.install\n",
    "```\n",
    "を実行した後、再度 Jupyter notebook を起動してください。"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "anaconda or miniconda は既にインストールされているものとします。"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "本ファイルのコマンドを順に実行した場合、カレントディレクトリ以下に解析作業用のフォルダが作られます。（解析教本の本文で示した手順とは異なります）"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`shift-Enter` を押すと選択中のセルの内容を実行することが可能です。"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##  解析ツールのインストール"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### FastQCのインストール"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "インストールの続行確認を省略するため `-y` オプションを指定して実行しています。"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "# All requested packages already installed.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "conda install -y -c bioconda fastqc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FastQC v0.11.8\n"
     ]
    }
   ],
   "source": [
    "fastqc -v"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### fastpのインストール"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /Users/nigyta/miniconda3\n",
      "\n",
      "  added / updated specs:\n",
      "    - fastp\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    fastp-0.20.0               |       hd9629dc_0         220 KB  bioconda\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:         220 KB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  fastp              bioconda/osx-64::fastp-0.20.0-hd9629dc_0\n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "fastp-0.20.0         | 220 KB    | ##################################### | 100% \n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n"
     ]
    }
   ],
   "source": [
    "conda install -y -c bioconda fastp"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "動作確認のためバージョンを表示"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "fastp 0.20.0\n"
     ]
    }
   ],
   "source": [
    "fastp -v"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### fastpのインストール"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /Users/nigyta/miniconda3\n",
      "\n",
      "  added / updated specs:\n",
      "    - seqkit\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    seqkit-0.11.0              |                0         6.7 MB  bioconda\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:         6.7 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  seqkit             bioconda/osx-64::seqkit-0.11.0-0\n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "seqkit-0.11.0        | 6.7 MB    | ##################################### | 100% \n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n"
     ]
    }
   ],
   "source": [
    "conda install -y -c bioconda seqkit"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "動作確認のためバージョンを表示"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "seqkit v0.11.0\n",
      "\n",
      "Checking new version...\n",
      "You are using the latest version of seqkit\n"
     ]
    }
   ],
   "source": [
    "seqkit version"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Platanus_Bのインストール"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "カレントディレクトリに`tools`ディレクトリを作成し、インストールを行います。Platanus_Bは[開発元ウェブサイト](http://platanus.bio.titech.ac.jp/platanus-b)から取得します。"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: tools: File exists\n",
      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
      "100 1407k  100 1407k    0     0   685k      0  0:00:02  0:00:02 --:--:--  685k\n",
      "x Platanus_B_v1.1.0_190607_macOS_bin/\n",
      "x Platanus_B_v1.1.0_190607_macOS_bin/platanus_b\n",
      "x Platanus_B_v1.1.0_190607_macOS_bin/README\n"
     ]
    }
   ],
   "source": [
    "mkdir tools\n",
    "cd tools\n",
    "curl http://platanus.bio.titech.ac.jp/?ddownload=411 --output platanus_tmp.tar.gz\n",
    "tar xvfz platanus_tmp.tar.gz\n",
    "cp Platanus_B_v1.1.0_190607_macOS_bin/platanus_b ./\n",
    "cp Platanus_B_v1.1.0_190607_macOS_bin/README ./platanus_README\n",
    "rm -rf Platanus_B_v1.1.0_190607_macOS_bin platanus_tmp.tar.gz\n",
    "cd ../"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "tools以下にインストールできたことを確認"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 10256\n",
      "-rw-r--r--  1 nigyta  staff     5587 12  2 21:44 platanus_README\n",
      "-rwxr-xr-x  1 nigyta  staff  4775076 12  2 21:44 platanus_b\n"
     ]
    }
   ],
   "source": [
    "ls -l tools/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "platanus_bのバージョン番号・ヘルプが表示されることを確認する。"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Platanus version: 1.1.0\n",
      "tools/platanus_b \n",
      "\n",
      "Usage: platanus_b Command [options]\n",
      "\n",
      "Command: assemble, iterate, scaffold, gap_close, polish, merge\n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "tools/platanus_b"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "後の操作を簡単にするため環境変数`PATH`に`tools`ディレクトリを追加。  \n",
    "（この操作をJupyter notebook以外から行う場合には、カレントディレクトリの場所によって正しく動作しない場合がありますのでご注意ください）"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "export PATH=$PWD/tools:$PATH"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "再度、動作確認を行い、platanus_bのバージョン番号・ヘルプが表示されることを確認する。"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Platanus version: 1.1.0\n",
      "platanus_b \n",
      "\n",
      "Usage: platanus_b Command [options]\n",
      "\n",
      "Command: assemble, iterate, scaffold, gap_close, polish, merge\n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "platanus_b"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## データの入手と前処理"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "以下の手順では、カレントディレクトリに作業フォルダとして`analysis`という名称のディレクトリを作成してディレクトリ内で解析を行います。"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/nigyta/bact_genome/analysis\n"
     ]
    }
   ],
   "source": [
    "mkdir analysis\n",
    "cd analysis\n",
    "pwd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`reads`というディレクトリを作り、その中にリード配列のデータを取得します。"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/nigyta/bact_genome/analysis/reads\n"
     ]
    }
   ],
   "source": [
    "mkdir reads\n",
    "cd reads\n",
    "pwd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__テスト用データの取得__"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "ネットワーク環境によっては数分かかります。"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
      "100  448M  100  448M    0     0  10.8M      0  0:00:41  0:00:41 --:--:-- 11.0M\n",
      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
      "100  503M  100  503M    0     0  10.5M      0  0:00:47  0:00:47 --:--:-- 11.1M\n"
     ]
    }
   ],
   "source": [
    "curl -O ftp://ftp.ddbj.nig.ac.jp//ddbj_database/dra/fastq/DRA002/DRA002643/DRX022186/DRR024501_1.fastq.bz2\n",
    "curl -O ftp://ftp.ddbj.nig.ac.jp//ddbj_database/dra/fastq/DRA002/DRA002643/DRX022186/DRR024501_2.fastq.bz2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "bunzip2 *bz2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__取得したデータの確認__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "file               format  type   num_seqs      sum_len  min_len  avg_len  max_len\n",
      "DRR024501_1.fastq  FASTQ   DNA   2,971,310  745,798,810      251      251      251\n",
      "DRR024501_2.fastq  FASTQ   DNA   2,971,310  745,798,810      251      251      251\n"
     ]
    }
   ],
   "source": [
    "seqkit stats *fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__FastQCの実行__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Started analysis of DRR024501_1.fastq\n",
      "Approx 5% complete for DRR024501_1.fastq\n",
      "Approx 10% complete for DRR024501_1.fastq\n",
      "Approx 15% complete for DRR024501_1.fastq\n",
      "Approx 20% complete for DRR024501_1.fastq\n",
      "Approx 25% complete for DRR024501_1.fastq\n",
      "Approx 30% complete for DRR024501_1.fastq\n",
      "Approx 35% complete for DRR024501_1.fastq\n",
      "Approx 40% complete for DRR024501_1.fastq\n",
      "Approx 45% complete for DRR024501_1.fastq\n",
      "Approx 50% complete for DRR024501_1.fastq\n",
      "Approx 55% complete for DRR024501_1.fastq\n",
      "Approx 60% complete for DRR024501_1.fastq\n",
      "Approx 65% complete for DRR024501_1.fastq\n",
      "Approx 70% complete for DRR024501_1.fastq\n",
      "Approx 75% complete for DRR024501_1.fastq\n",
      "Approx 80% complete for DRR024501_1.fastq\n",
      "Approx 85% complete for DRR024501_1.fastq\n",
      "Approx 90% complete for DRR024501_1.fastq\n",
      "Approx 95% complete for DRR024501_1.fastq\n",
      "Analysis complete for DRR024501_1.fastq\n",
      "Started analysis of DRR024501_2.fastq\n",
      "Approx 5% complete for DRR024501_2.fastq\n",
      "Approx 10% complete for DRR024501_2.fastq\n",
      "Approx 15% complete for DRR024501_2.fastq\n",
      "Approx 20% complete for DRR024501_2.fastq\n",
      "Approx 25% complete for DRR024501_2.fastq\n",
      "Approx 30% complete for DRR024501_2.fastq\n",
      "Approx 35% complete for DRR024501_2.fastq\n",
      "Approx 40% complete for DRR024501_2.fastq\n",
      "Approx 45% complete for DRR024501_2.fastq\n",
      "Approx 50% complete for DRR024501_2.fastq\n",
      "Approx 55% complete for DRR024501_2.fastq\n",
      "Approx 60% complete for DRR024501_2.fastq\n",
      "Approx 65% complete for DRR024501_2.fastq\n",
      "Approx 70% complete for DRR024501_2.fastq\n",
      "Approx 75% complete for DRR024501_2.fastq\n",
      "Approx 80% complete for DRR024501_2.fastq\n",
      "Approx 85% complete for DRR024501_2.fastq\n",
      "Approx 90% complete for DRR024501_2.fastq\n",
      "Approx 95% complete for DRR024501_2.fastq\n",
      "Analysis complete for DRR024501_2.fastq\n"
     ]
    }
   ],
   "source": [
    "fastqc *fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__アダプター配列の除去__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Read1 before filtering:\n",
      "total reads: 2971310\n",
      "total bases: 745798810\n",
      "Q20 bases: 696510031(93.3911%)\n",
      "Q30 bases: 651512094(87.3576%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 2971310\n",
      "total bases: 745798810\n",
      "Q20 bases: 630071152(84.4827%)\n",
      "Q30 bases: 554509571(74.3511%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 2779208\n",
      "total bases: 676268617\n",
      "Q20 bases: 647164103(95.6963%)\n",
      "Q30 bases: 612568020(90.5806%)\n",
      "\n",
      "Read2 aftering filtering:\n",
      "total reads: 2779208\n",
      "total bases: 676268617\n",
      "Q20 bases: 599957379(88.7158%)\n",
      "Q30 bases: 536918684(79.3943%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 5558416\n",
      "reads failed due to low quality: 369366\n",
      "reads failed due to too many N: 14838\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 602436\n",
      "bases trimmed due to adapters: 43092538\n",
      "\n",
      "Duplication rate: 0.421494%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 383\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -i DRR024501_1.fastq -I DRR024501_2.fastq -o DRR024501_1.fastp.fastq -O DRR024501_2.fastp.fastq \n",
      "fastp v0.20.0, time used: 85 seconds\n"
     ]
    }
   ],
   "source": [
    "fastp -i DRR024501_1.fastq -I DRR024501_2.fastq -o DRR024501_1.fastp.fastq -O DRR024501_2.fastp.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__再度FastQCを実行__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Started analysis of DRR024501_1.fastp.fastq\n",
      "Approx 5% complete for DRR024501_1.fastp.fastq\n",
      "Approx 10% complete for DRR024501_1.fastp.fastq\n",
      "Approx 15% complete for DRR024501_1.fastp.fastq\n",
      "Approx 20% complete for DRR024501_1.fastp.fastq\n",
      "Approx 25% complete for DRR024501_1.fastp.fastq\n",
      "Approx 30% complete for DRR024501_1.fastp.fastq\n",
      "Approx 35% complete for DRR024501_1.fastp.fastq\n",
      "Approx 40% complete for DRR024501_1.fastp.fastq\n",
      "Approx 45% complete for DRR024501_1.fastp.fastq\n",
      "Approx 50% complete for DRR024501_1.fastp.fastq\n",
      "Approx 55% complete for DRR024501_1.fastp.fastq\n",
      "Approx 60% complete for DRR024501_1.fastp.fastq\n",
      "Approx 65% complete for DRR024501_1.fastp.fastq\n",
      "Approx 70% complete for DRR024501_1.fastp.fastq\n",
      "Approx 75% complete for DRR024501_1.fastp.fastq\n",
      "Approx 80% complete for DRR024501_1.fastp.fastq\n",
      "Approx 85% complete for DRR024501_1.fastp.fastq\n",
      "Approx 90% complete for DRR024501_1.fastp.fastq\n",
      "Approx 95% complete for DRR024501_1.fastp.fastq\n",
      "Analysis complete for DRR024501_1.fastp.fastq\n",
      "Started analysis of DRR024501_2.fastp.fastq\n",
      "Approx 5% complete for DRR024501_2.fastp.fastq\n",
      "Approx 10% complete for DRR024501_2.fastp.fastq\n",
      "Approx 15% complete for DRR024501_2.fastp.fastq\n",
      "Approx 20% complete for DRR024501_2.fastp.fastq\n",
      "Approx 25% complete for DRR024501_2.fastp.fastq\n",
      "Approx 30% complete for DRR024501_2.fastp.fastq\n",
      "Approx 35% complete for DRR024501_2.fastp.fastq\n",
      "Approx 40% complete for DRR024501_2.fastp.fastq\n",
      "Approx 45% complete for DRR024501_2.fastp.fastq\n",
      "Approx 50% complete for DRR024501_2.fastp.fastq\n",
      "Approx 55% complete for DRR024501_2.fastp.fastq\n",
      "Approx 60% complete for DRR024501_2.fastp.fastq\n",
      "Approx 65% complete for DRR024501_2.fastp.fastq\n",
      "Approx 70% complete for DRR024501_2.fastp.fastq\n",
      "Approx 75% complete for DRR024501_2.fastp.fastq\n",
      "Approx 80% complete for DRR024501_2.fastp.fastq\n",
      "Approx 85% complete for DRR024501_2.fastp.fastq\n",
      "Approx 90% complete for DRR024501_2.fastp.fastq\n",
      "Approx 95% complete for DRR024501_2.fastp.fastq\n",
      "Analysis complete for DRR024501_2.fastp.fastq\n"
     ]
    }
   ],
   "source": [
    "fastqc *.fastp.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__fastp 処理結果の確認__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "file                     format  type   num_seqs      sum_len  min_len  avg_len  max_len\n",
      "DRR024501_1.fastp.fastq  FASTQ   DNA   2,779,208  676,268,617       31    243.3      251\n",
      "DRR024501_1.fastq        FASTQ   DNA   2,971,310  745,798,810      251      251      251\n",
      "DRR024501_2.fastp.fastq  FASTQ   DNA   2,779,208  676,268,617       31    243.3      251\n",
      "DRR024501_2.fastq        FASTQ   DNA   2,971,310  745,798,810      251      251      251\n"
     ]
    }
   ],
   "source": [
    "seqkit stats *fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__データのサブサンプリング__"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "カバレッジが約100xになるよう、全データのうち20%を抽出する。（ゲノムサイズは約2.5Mbpと想定）"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[INFO]\u001b[0m sample by proportion\n",
      "[INFO]\u001b[0m 556074 sequences outputted\n",
      "[INFO]\u001b[0m sample by proportion\n",
      "[INFO]\u001b[0m 556074 sequences outputted\n"
     ]
    }
   ],
   "source": [
    "seqkit sample -p 0.2 DRR024501_1.fastp.fastq > DRR024501_1.sampled.fastp.fastq\n",
    "seqkit sample -p 0.2 DRR024501_2.fastp.fastq > DRR024501_2.sampled.fastp.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__サブサンプリングしたデータの確認__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "seqkit stats *.sampled.fastp.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## アセンブリ"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "ディレクトリの準備  \n",
    "`assembly`という名称で作業ディレクトリを作成する。"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/nigyta/bact_genome/analysis/assembly\n"
     ]
    }
   ],
   "source": [
    "cd ../\n",
    "mkdir assembly\n",
    "cd assembly\n",
    "pwd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "（確認）リードデータは、一つ上の階層にあるreadsディレクトリに格納されている"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 15816912\n",
      "-rw-r--r--  1 nigyta  staff  1770380254 12  2 22:06 DRR024501_1.fastp.fastq\n",
      "-rw-r--r--  1 nigyta  staff      231274 12  2 22:06 DRR024501_1.fastp_fastqc.html\n",
      "-rw-r--r--  1 nigyta  staff      251019 12  2 22:06 DRR024501_1.fastp_fastqc.zip\n",
      "-rw-r--r--  1 nigyta  staff  1938304424 12  2 21:56 DRR024501_1.fastq\n",
      "-rw-r--r--  1 nigyta  staff   314019940 12  2 22:07 DRR024501_1.sampled.fastp.fastq\n",
      "-rw-r--r--  1 nigyta  staff      239558 12  2 22:03 DRR024501_1_fastqc.html\n",
      "-rw-r--r--  1 nigyta  staff      268951 12  2 22:03 DRR024501_1_fastqc.zip\n",
      "-rw-r--r--  1 nigyta  staff  1770380254 12  2 22:06 DRR024501_2.fastp.fastq\n",
      "-rw-r--r--  1 nigyta  staff      231004 12  2 22:07 DRR024501_2.fastp_fastqc.html\n",
      "-rw-r--r--  1 nigyta  staff      253422 12  2 22:07 DRR024501_2.fastp_fastqc.zip\n",
      "-rw-r--r--  1 nigyta  staff  1938304424 12  2 21:56 DRR024501_2.fastq\n",
      "-rw-r--r--  1 nigyta  staff   314019940 12  2 22:08 DRR024501_2.sampled.fastp.fastq\n",
      "-rw-r--r--  1 nigyta  staff      244115 12  2 22:04 DRR024501_2_fastqc.html\n",
      "-rw-r--r--  1 nigyta  staff      277506 12  2 22:04 DRR024501_2_fastqc.zip\n",
      "-rw-r--r--  1 nigyta  staff      545747 12  2 22:06 fastp.html\n",
      "-rw-r--r--  1 nigyta  staff      174310 12  2 22:06 fastp.json\n"
     ]
    }
   ],
   "source": [
    "ls -l ../reads/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__アセンブリ1 (platanus_b の assemble ステップ)__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Platanus version: 1.1.0\n",
      "platanus_b assemble -t 2 -f ../reads/DRR024501_1.sampled.fastp.fastq ../reads/DRR024501_2.sampled.fastp.fastq \n",
      "\n",
      "K = 32, saving kmers from reads...\n",
      "AVE_READ_LEN=243.268\n",
      "\n",
      "KMER_EXTENSION:\n",
      "K=32, KMER_COVERAGE=90.322 (>= 7), COVERAGE_CUTOFF=7\n",
      "K=42, KMER_COVERAGE=86.0669, COVERAGE_CUTOFF=7, PROB_SPLIT=10e-inf\n",
      "K=52, KMER_COVERAGE=81.8118, COVERAGE_CUTOFF=7, PROB_SPLIT=10e-inf\n",
      "K=62, KMER_COVERAGE=77.5567, COVERAGE_CUTOFF=7, PROB_SPLIT=10e-inf\n",
      "K=72, KMER_COVERAGE=73.3016, COVERAGE_CUTOFF=7, PROB_SPLIT=10e-inf\n",
      "K=82, KMER_COVERAGE=69.0465, COVERAGE_CUTOFF=7, PROB_SPLIT=10e-inf\n",
      "K=92, KMER_COVERAGE=64.7914, COVERAGE_CUTOFF=7, PROB_SPLIT=10e-inf\n",
      "K=102, KMER_COVERAGE=60.5363, COVERAGE_CUTOFF=7, PROB_SPLIT=10e-inf\n",
      "K=112, KMER_COVERAGE=56.2812, COVERAGE_CUTOFF=7, PROB_SPLIT=10e-15.6536\n",
      "K=122, KMER_COVERAGE=52.0261, COVERAGE_CUTOFF=7, PROB_SPLIT=10e-14.0625\n",
      "K=132, KMER_COVERAGE=47.771, COVERAGE_CUTOFF=7, PROB_SPLIT=10e-12.4307\n",
      "K=142, KMER_COVERAGE=43.5159, COVERAGE_CUTOFF=7, PROB_SPLIT=10e-10.8201\n",
      "K=152, KMER_COVERAGE=39.2608, COVERAGE_CUTOFF=6, PROB_SPLIT=10e-10.0613\n",
      "K=162, KMER_COVERAGE=35.0057, COVERAGE_CUTOFF=4, PROB_SPLIT=10e-10.2694\n",
      "K=172, KMER_COVERAGE=30.7506, COVERAGE_CUTOFF=2, PROB_SPLIT=10e-10.8117\n",
      "K=182, KMER_COVERAGE=26.4955, COVERAGE_CUTOFF=1, PROB_SPLIT=10e-10.4655\n",
      "K=186, KMER_COVERAGE=24.7935, COVERAGE_CUTOFF=1, PROB_SPLIT=10e-10.0687\n",
      "K=187, KMER_COVERAGE=24.368, COVERAGE_CUTOFF=1, PROB_SPLIT=10e-10.2818\n",
      "K=188, KMER_COVERAGE=23.9425, COVERAGE_CUTOFF=1, PROB_SPLIT=10e-10.097\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=814\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=814\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=78\n",
      "NUM_REMOVED_NODES=1\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=79\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 42, loading kmers from contigs...\n",
      "K = 42, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 7\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=99\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=99\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=29\n",
      "NUM_REMOVED_NODES=1\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=30\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 52, loading kmers from contigs...\n",
      "K = 52, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 7\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=77\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=77\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=21\n",
      "NUM_REMOVED_NODES=1\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=22\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 62, loading kmers from contigs...\n",
      "K = 62, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 7\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=58\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=58\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=21\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=21\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 72, loading kmers from contigs...\n",
      "K = 72, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 7\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=44\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=44\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=20\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=20\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 82, loading kmers from contigs...\n",
      "K = 82, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 7\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=43\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=43\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=16\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=16\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 92, loading kmers from contigs...\n",
      "K = 92, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 7\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=44\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=44\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=20\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=20\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 102, loading kmers from contigs...\n",
      "K = 102, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 7\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=38\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=38\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=16\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=16\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 112, loading kmers from contigs...\n",
      "K = 112, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 7\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=36\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=36\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=15\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=15\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 122, loading kmers from contigs...\n",
      "K = 122, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 7\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=39\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=39\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=13\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=13\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 132, loading kmers from contigs...\n",
      "K = 132, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 7\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=35\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=35\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=8\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=8\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 142, loading kmers from contigs...\n",
      "K = 142, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 7\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=36\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=36\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=3\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=3\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 152, loading kmers from contigs...\n",
      "K = 152, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 6\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=33\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=33\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=4\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=4\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 162, loading kmers from contigs...\n",
      "K = 162, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 4\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=65\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=65\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=9\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=9\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 172, loading kmers from contigs...\n",
      "K = 172, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 2\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=396\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=396\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=35\n",
      "NUM_REMOVED_NODES=2\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=37\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 182, loading kmers from contigs...\n",
      "K = 182, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 1\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=3096\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=3096\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=253\n",
      "NUM_REMOVED_NODES=17\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=270\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 186, loading kmers from contigs...\n",
      "K = 186, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 1\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=2322\n",
      "NUM_REMOVED_NODES=2\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=2324\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=181\n",
      "NUM_REMOVED_NODES=10\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=191\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 187, loading kmers from contigs...\n",
      "K = 187, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 1\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=1837\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=1837\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=137\n",
      "NUM_REMOVED_NODES=5\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=142\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 188, loading kmers from contigs...\n",
      "K = 188, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 1\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=1698\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=1698\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=125\n",
      "NUM_REMOVED_NODES=4\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=129\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "LENGTH_CUTOFF = 376\n",
      "COVERAGE_CUTOFF = 13\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=2\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=2\n",
      "AVE_KMER_COV_REMOVING_BUBBLE=23.9671\n",
      "K = 32, saving kmers from reads...\n",
      "AVE_READ_LEN=243.268\n",
      "\n",
      "KMER_EXTENSION:\n",
      "K=32, KMER_COVERAGE=108.291 (>= 90), COVERAGE_CUTOFF=90\n",
      "K=42, KMER_COVERAGE=103.19, COVERAGE_CUTOFF=43, PROB_SPLIT=10e-10.1259\n",
      "K=52, KMER_COVERAGE=98.0881, COVERAGE_CUTOFF=39, PROB_SPLIT=10e-10.3863\n",
      "K=62, KMER_COVERAGE=92.9864, COVERAGE_CUTOFF=36, PROB_SPLIT=10e-10.2608\n",
      "K=72, KMER_COVERAGE=87.8848, COVERAGE_CUTOFF=33, PROB_SPLIT=10e-10.149\n",
      "K=82, KMER_COVERAGE=82.7832, COVERAGE_CUTOFF=30, PROB_SPLIT=10e-10.0539\n",
      "K=92, KMER_COVERAGE=77.6815, COVERAGE_CUTOFF=26, PROB_SPLIT=10e-10.463\n",
      "K=102, KMER_COVERAGE=72.5799, COVERAGE_CUTOFF=23, PROB_SPLIT=10e-10.4391\n",
      "K=112, KMER_COVERAGE=67.4782, COVERAGE_CUTOFF=20, PROB_SPLIT=10e-10.4547\n",
      "K=122, KMER_COVERAGE=62.3766, COVERAGE_CUTOFF=17, PROB_SPLIT=10e-10.523\n",
      "K=132, KMER_COVERAGE=57.2749, COVERAGE_CUTOFF=15, PROB_SPLIT=10e-10.043\n",
      "K=142, KMER_COVERAGE=52.1733, COVERAGE_CUTOFF=12, PROB_SPLIT=10e-10.2262\n",
      "K=152, KMER_COVERAGE=47.0717, COVERAGE_CUTOFF=9, PROB_SPLIT=10e-10.5463\n",
      "K=162, KMER_COVERAGE=41.97, COVERAGE_CUTOFF=7, PROB_SPLIT=10e-10.2405\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "K=172, KMER_COVERAGE=36.8684, COVERAGE_CUTOFF=5, PROB_SPLIT=10e-10.0356\n",
      "K=182, KMER_COVERAGE=31.7667, COVERAGE_CUTOFF=3, PROB_SPLIT=10e-10.0245\n",
      "K=192, KMER_COVERAGE=26.6651, COVERAGE_CUTOFF=1, PROB_SPLIT=10e-10.5391\n",
      "K=195, KMER_COVERAGE=25.1346, COVERAGE_CUTOFF=1, PROB_SPLIT=10e-10.3138\n",
      "K=196, KMER_COVERAGE=24.6244, COVERAGE_CUTOFF=1, PROB_SPLIT=10e-10.3932\n",
      "K=197, KMER_COVERAGE=24.1143, COVERAGE_CUTOFF=1, PROB_SPLIT=10e-10.1717\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=23486\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=23486\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 42, loading kmers from contigs...\n",
      "K = 42, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 43\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=316\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=316\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=197\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=197\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 52, loading kmers from contigs...\n",
      "K = 52, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 39\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=181\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=181\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=135\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=135\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 62, loading kmers from contigs...\n",
      "K = 62, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 36\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=167\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=167\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=109\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=109\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 72, loading kmers from contigs...\n",
      "K = 72, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 33\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=172\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=172\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=79\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=79\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 82, loading kmers from contigs...\n",
      "K = 82, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 30\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=171\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=171\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=64\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=64\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 92, loading kmers from contigs...\n",
      "K = 92, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 26\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=141\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=141\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=35\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=35\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 102, loading kmers from contigs...\n",
      "K = 102, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 23\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=117\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=117\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=27\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=27\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 112, loading kmers from contigs...\n",
      "K = 112, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 20\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=124\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=124\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=7\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=7\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 122, loading kmers from contigs...\n",
      "K = 122, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 17\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=129\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=129\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 132, loading kmers from contigs...\n",
      "K = 132, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 15\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=109\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=109\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=6\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=6\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "K = 142, loading kmers from contigs...\n",
      "K = 142, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 12\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=100\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=100\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=17\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=17\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 152, loading kmers from contigs...\n",
      "K = 152, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 9\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=69\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=69\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=35\n",
      "NUM_REMOVED_NODES=1\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=36\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 162, loading kmers from contigs...\n",
      "K = 162, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 7\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=63\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=63\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=34\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=34\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 172, loading kmers from contigs...\n",
      "K = 172, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 5\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=57\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=57\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=42\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=42\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 182, loading kmers from contigs...\n",
      "K = 182, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 3\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=81\n",
      "NUM_REMOVED_NODES=1\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=82\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=50\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=50\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 192, loading kmers from contigs...\n",
      "K = 192, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 1\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=3683\n",
      "NUM_REMOVED_NODES=4\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=3687\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=202\n",
      "NUM_REMOVED_NODES=4\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=206\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 195, loading kmers from contigs...\n",
      "K = 195, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 1\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=1848\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=1848\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=146\n",
      "NUM_REMOVED_NODES=6\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=152\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 196, loading kmers from contigs...\n",
      "K = 196, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 1\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=1463\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=1463\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=114\n",
      "NUM_REMOVED_NODES=6\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=120\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 197, loading kmers from contigs...\n",
      "K = 197, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 1\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=1379\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=1379\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=107\n",
      "NUM_REMOVED_NODES=6\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=113\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "LENGTH_CUTOFF = 394\n",
      "COVERAGE_CUTOFF = 13\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=0\n",
      "AVE_KMER_COV_REMOVING_BUBBLE=25.937\n",
      "K = 32, saving kmers from reads...\n",
      "AVE_READ_LEN=243.268\n",
      "\n",
      "KMER_EXTENSION:\n",
      "K=32, KMER_COVERAGE=315.981 (>= 180), COVERAGE_CUTOFF=180\n",
      "K=42, KMER_COVERAGE=301.095, COVERAGE_CUTOFF=180, PROB_SPLIT=10e-12.6952\n",
      "K=52, KMER_COVERAGE=286.209, COVERAGE_CUTOFF=180, PROB_SPLIT=10e-10.139\n",
      "K=62, KMER_COVERAGE=271.323, COVERAGE_CUTOFF=168, PROB_SPLIT=10e-10.1688\n",
      "K=72, KMER_COVERAGE=256.437, COVERAGE_CUTOFF=156, PROB_SPLIT=10e-10.2195\n",
      "K=82, KMER_COVERAGE=241.551, COVERAGE_CUTOFF=145, PROB_SPLIT=10e-10.0664\n",
      "K=92, KMER_COVERAGE=226.665, COVERAGE_CUTOFF=133, PROB_SPLIT=10e-10.1627\n",
      "K=102, KMER_COVERAGE=211.779, COVERAGE_CUTOFF=122, PROB_SPLIT=10e-10.0476\n",
      "K=112, KMER_COVERAGE=196.893, COVERAGE_CUTOFF=110, PROB_SPLIT=10e-10.2122\n",
      "K=122, KMER_COVERAGE=182.007, COVERAGE_CUTOFF=99, PROB_SPLIT=10e-10.159\n",
      "K=132, KMER_COVERAGE=167.121, COVERAGE_CUTOFF=88, PROB_SPLIT=10e-10.1448\n",
      "K=142, KMER_COVERAGE=152.235, COVERAGE_CUTOFF=77, PROB_SPLIT=10e-10.1829\n",
      "K=152, KMER_COVERAGE=137.349, COVERAGE_CUTOFF=66, PROB_SPLIT=10e-10.293\n",
      "K=162, KMER_COVERAGE=122.464, COVERAGE_CUTOFF=56, PROB_SPLIT=10e-10.1532\n",
      "K=172, KMER_COVERAGE=107.578, COVERAGE_CUTOFF=46, PROB_SPLIT=10e-10.0987\n",
      "K=182, KMER_COVERAGE=92.6915, COVERAGE_CUTOFF=36, PROB_SPLIT=10e-10.1802\n",
      "K=192, KMER_COVERAGE=77.8056, COVERAGE_CUTOFF=27, PROB_SPLIT=10e-10.0158\n",
      "K=202, KMER_COVERAGE=62.9196, COVERAGE_CUTOFF=18, PROB_SPLIT=10e-10.1225\n",
      "K=212, KMER_COVERAGE=48.0336, COVERAGE_CUTOFF=10, PROB_SPLIT=10e-10.1575\n",
      "K=222, KMER_COVERAGE=33.1476, COVERAGE_CUTOFF=3, PROB_SPLIT=10e-10.5884\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "K=227, KMER_COVERAGE=25.7046, COVERAGE_CUTOFF=1, PROB_SPLIT=10e-10.3852\n",
      "K=228, KMER_COVERAGE=24.216, COVERAGE_CUTOFF=1, PROB_SPLIT=10e-10.2159\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=578\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=578\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 42, loading kmers from contigs...\n",
      "K = 42, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 180\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=9\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=9\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=52\n",
      "NUM_REMOVED_NODES=1\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=53\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 52, loading kmers from contigs...\n",
      "K = 52, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 180\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=58\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=58\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 62, loading kmers from contigs...\n",
      "K = 62, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 168\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=37\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=37\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 72, loading kmers from contigs...\n",
      "K = 72, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 156\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=43\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=43\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 82, loading kmers from contigs...\n",
      "K = 82, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 145\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=2\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=2\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=26\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=26\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 92, loading kmers from contigs...\n",
      "K = 92, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 133\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=2\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=2\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=20\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=20\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 102, loading kmers from contigs...\n",
      "K = 102, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 122\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=3\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=3\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=17\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=17\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 112, loading kmers from contigs...\n",
      "K = 112, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 110\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=6\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=6\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=12\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=12\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 122, loading kmers from contigs...\n",
      "K = 122, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 99\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=5\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=5\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=11\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=11\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 132, loading kmers from contigs...\n",
      "K = 132, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 88\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=5\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=5\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=12\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=12\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 142, loading kmers from contigs...\n",
      "K = 142, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 77\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=4\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=4\n",
      "removing branches...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=11\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=11\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 152, loading kmers from contigs...\n",
      "K = 152, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 66\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=7\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=7\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=10\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=10\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 162, loading kmers from contigs...\n",
      "K = 162, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 56\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=8\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=8\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=4\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=4\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 172, loading kmers from contigs...\n",
      "K = 172, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 46\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=6\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=6\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=1\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=1\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 182, loading kmers from contigs...\n",
      "K = 182, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 36\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=6\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=6\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 192, loading kmers from contigs...\n",
      "K = 192, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 27\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=8\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=8\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 202, loading kmers from contigs...\n",
      "K = 202, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 18\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=27\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=27\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 212, loading kmers from contigs...\n",
      "K = 212, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 10\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=29\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=29\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=1\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=1\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 222, loading kmers from contigs...\n",
      "K = 222, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 3\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=14\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=14\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=23\n",
      "NUM_REMOVED_NODES=1\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=24\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 227, loading kmers from contigs...\n",
      "K = 227, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 1\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=86\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=86\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=33\n",
      "NUM_REMOVED_NODES=1\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=34\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "extracting reads (containing kmer used in contig assemble)...\n",
      "K = 228, loading kmers from contigs...\n",
      "K = 228, saving additional kmers(not found in contigs) from reads...\n",
      "COVERAGE_CUTOFF = 1\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=63\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=63\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=30\n",
      "NUM_REMOVED_NODES=1\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=31\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "LENGTH_CUTOFF = 456\n",
      "COVERAGE_CUTOFF = 16\n",
      "removing erroneous nodes...\n",
      "NUM_REMOVED_NODES=1\n",
      "NUM_REMOVED_NODES=0\n",
      "TOTAL_NUM_REMOVED_NODES=1\n",
      "AVE_KMER_COV_REMOVING_BUBBLE=29.4726\n",
      "K = 228, loading kmers from contigs...\n",
      "loading kmers...\n",
      "connecting kmers...\n",
      "removing branches...\n",
      "BRANCH_DELETE_THRESHOLD=0.5\n",
      "NUM_CUT=0\n",
      "TOTAL_NUM_CUT=0\n",
      "removing bubbles...\n",
      "BUBBLE_IDENTITY_THRESHOLD=0\n",
      "NUM_REMOVED_BUBBLES=0\n",
      "TOTAL_NUM_REMOVED_BUBBLES=0\n",
      "assemble completed!\n",
      "\n",
      "#### PROCESS INFORMATION ####\n",
      "VmPeak:           0.000 GByte\n",
      "VmHWM:            0.000 GByte\n"
     ]
    }
   ],
   "source": [
    "platanus_b assemble -t 2 -f ../reads/DRR024501_1.sampled.fastp.fastq ../reads/DRR024501_2.sampled.fastp.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "上記コマンドが正しく実行できない場合、`tools`ディレクトリに`PATH`が通っていること、および、\n",
    "```\n",
    "ls ../reads/DRR024501_1.sampled.fastp.fastq ../reads/DRR024501_2.sampled.fastp.fastq\n",
    "```\n",
    "を実行してみて`reads`ディレクトリにファイルが存在していることを確認すること。"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "アセンブリ結果 (out_contig.fa) の確認"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "file           format  type  num_seqs    sum_len  min_len  avg_len  max_len   Q1   Q2     Q3  sum_gap     N50  Q20(%)  Q30(%)\n",
      "out_contig.fa  FASTA   DNA        369  2,445,694      228  6,627.9   88,589  245  393  7,138        0  28,870       0       0\n"
     ]
    }
   ],
   "source": [
    "seqkit stats -a -G N out_contig.fa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__アセンブリ2 (platanus_b の iterate ステップ)__"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "実行には数10分程度かかる。"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Platanus version: 1.1.0\n",
      "platanus_b iterate -t 2 -c out_contig.fa -IP1 ../reads/DRR024501_1.sampled.fastp.fastq ../reads/DRR024501_2.sampled.fastp.fastq \n",
      "\n",
      "\n",
      "#### PROCESS INFORMATION ####\n",
      "VmPeak:           0.000 GByte\n",
      "VmHWM:            0.000 GByte\n"
     ]
    }
   ],
   "source": [
    "platanus_b iterate -t 2 -c out_contig.fa -IP1 ../reads/DRR024501_1.sampled.fastp.fastq ../reads/DRR024501_2.sampled.fastp.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "アセンブリ結果 (out_iterativeAssembly.fa) の確認"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "file                      format  type  num_seqs    sum_len  min_len   avg_len  max_len   Q1      Q2      Q3  sum_gap     N50  Q20(%)  Q30(%)\n",
      "out_iterativeAssembly.fa  FASTA   DNA         55  2,365,587      246  43,010.7  257,852  709  14,430  69,725    2,619  96,297       0       0\n"
     ]
    }
   ],
   "source": [
    "seqkit stats -a -G N out_iterativeAssembly.fa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## アノテーション"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "本文ではアセンブリ結果ファイル (out_iterativeAssembly.fa) を [DFAST ウェブサービス](https://dfast.nig.ac.jp) にアップロードしてアノテーションを行っています。  \n",
    "本文執筆後、DFAST のローカル版が bioconda でインストールできるようになったため、下記ではローカル版のインストールおよび利用方法を紹介します。\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### DFAST ローカル版のインストール"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "インストール時に参照データベースファイルの取得等を行うため、ネットワーク環境によっては数分〜10分程度かかります。"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /Users/nigyta/miniconda3\n",
      "\n",
      "  added / updated specs:\n",
      "    - dfast\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    aragorn-1.2.38             |                2         134 KB  bioconda\n",
      "    barrnap-0.9                |                3         556 KB  bioconda\n",
      "    bedtools-2.29.0            |       h37cfd92_3         860 KB  bioconda\n",
      "    biopython-1.74             |   py37h1de35cc_0         2.0 MB\n",
      "    blas-1.0                   |              mkl           6 KB\n",
      "    blast-2.6.0                |      boost1.64_2       113.0 MB  bioconda\n",
      "    bzip2-1.0.8                |       h1de35cc_0          75 KB\n",
      "    dfast-1.2.3                |      py37pl526_0        12.8 MB  bioconda\n",
      "    ghostx-1.3.7               |                0         274 KB  bioconda\n",
      "    hmmer-3.2.1                |       h6de7cb9_2         6.8 MB  bioconda\n",
      "    intel-openmp-2019.4        |              233         887 KB\n",
      "    libgcc-4.8.5               |      hdbeacc1_10         250 KB\n",
      "    libgfortran-3.0.1          |       h93005f0_2         426 KB\n",
      "    metagene_annotator-1.0     |       h01d97ff_3         350 KB  bioconda\n",
      "    mkl-2019.4                 |              233       101.9 MB\n",
      "    mkl-service-2.3.0          |   py37hfbe908c_0         201 KB\n",
      "    mkl_fft-1.0.15             |   py37h5e564d8_0         157 KB\n",
      "    mkl_random-1.1.0           |   py37ha771720_0         275 KB\n",
      "    numpy-1.17.4               |   py37h890c691_0           4 KB\n",
      "    numpy-base-1.17.4          |   py37h6575580_0         3.9 MB\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:       244.8 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  aragorn            bioconda/osx-64::aragorn-1.2.38-2\n",
      "  barrnap            bioconda/noarch::barrnap-0.9-3\n",
      "  bedtools           bioconda/osx-64::bedtools-2.29.0-h37cfd92_3\n",
      "  biopython          pkgs/main/osx-64::biopython-1.74-py37h1de35cc_0\n",
      "  blas               pkgs/main/osx-64::blas-1.0-mkl\n",
      "  blast              bioconda/osx-64::blast-2.6.0-boost1.64_2\n",
      "  bzip2              pkgs/main/osx-64::bzip2-1.0.8-h1de35cc_0\n",
      "  dfast              bioconda/noarch::dfast-1.2.3-py37pl526_0\n",
      "  ghostx             bioconda/osx-64::ghostx-1.3.7-0\n",
      "  hmmer              bioconda/osx-64::hmmer-3.2.1-h6de7cb9_2\n",
      "  intel-openmp       pkgs/main/osx-64::intel-openmp-2019.4-233\n",
      "  libgcc             pkgs/main/osx-64::libgcc-4.8.5-hdbeacc1_10\n",
      "  libgfortran        pkgs/main/osx-64::libgfortran-3.0.1-h93005f0_2\n",
      "  metagene_annotator bioconda/osx-64::metagene_annotator-1.0-h01d97ff_3\n",
      "  mkl                pkgs/main/osx-64::mkl-2019.4-233\n",
      "  mkl-service        pkgs/main/osx-64::mkl-service-2.3.0-py37hfbe908c_0\n",
      "  mkl_fft            pkgs/main/osx-64::mkl_fft-1.0.15-py37h5e564d8_0\n",
      "  mkl_random         pkgs/main/osx-64::mkl_random-1.1.0-py37ha771720_0\n",
      "  numpy              pkgs/main/osx-64::numpy-1.17.4-py37h890c691_0\n",
      "  numpy-base         pkgs/main/osx-64::numpy-base-1.17.4-py37h6575580_0\n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "metagene_annotator-1 | 350 KB    | ##################################### | 100% \n",
      "libgfortran-3.0.1    | 426 KB    | ##################################### | 100% \n",
      "bedtools-2.29.0      | 860 KB    | ##################################### | 100% \n",
      "biopython-1.74       | 2.0 MB    | ##################################### | 100% \n",
      "libgcc-4.8.5         | 250 KB    | ##################################### | 100% \n",
      "numpy-1.17.4         | 4 KB      | ##################################### | 100% \n",
      "bzip2-1.0.8          | 75 KB     | ##################################### | 100% \n",
      "mkl_fft-1.0.15       | 157 KB    | ##################################### | 100% \n",
      "dfast-1.2.3          | 12.8 MB   | ##################################### | 100% \n",
      "intel-openmp-2019.4  | 887 KB    | ##################################### | 100% \n",
      "mkl-service-2.3.0    | 201 KB    | ##################################### | 100% \n",
      "numpy-base-1.17.4    | 3.9 MB    | ##################################### | 100% \n",
      "barrnap-0.9          | 556 KB    | ##################################### | 100% \n",
      "aragorn-1.2.38       | 134 KB    | ##################################### | 100% \n",
      "ghostx-1.3.7         | 274 KB    | ##################################### | 100% \n",
      "mkl-2019.4           | 101.9 MB  | ##################################### | 100% \n",
      "mkl_random-1.1.0     | 275 KB    | ##################################### | 100% \n",
      "hmmer-3.2.1          | 6.8 MB    | ##################################### | 100% \n",
      "blas-1.0             | 6 KB      | ##################################### | 100% \n",
      "blast-2.6.0          | 113.0 MB  | ##################################### | 100% \n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n"
     ]
    }
   ],
   "source": [
    "conda install -y -c bioconda dfast"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "動作確認"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "dfast -v"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### DFASTの実行"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "初回実行時には参照データベースファイルの indexing を行うため時間がかかる。二回目以降は実行時間が短縮される。"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "dfast --genome out_iterativeAssembly.fa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "結果ファイルは `OUT` ディレクトリ内に出力される。出力ファイルには、\n",
    "- genome.gbk (GenBank 形式のアノテーション結果ファイル)\n",
    "- genome.gff (GFF 形式のアノテーション結果ファイル)\n",
    "- protein.faa (FASTA 形式のアノテーションされた遺伝子のアミノ酸配列ファイル)\n",
    "- cds.fna (FASTA 形式のアノテーションされた遺伝子の CDS 塩基配列ファイル)\n",
    "\n",
    "などが含まれる。"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "詳細な使用方法はヘルプ `dfast -h` や、[公式サイト](https://github.com/nigyta/dfast_core)のドキュメントをご参照ください。"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
